src/userflow.ts

```diff
@@ -206,6 +207,30 @@ interface Deferred {
 // empty object instead of `window`.
 var w: WindowWithUserflow = typeof window === 'undefined' ? ({} as any) : window
 var userflow = w.userflow
+
+// indicates if the history API's pushState and replaceState are patched with custom event emitters
+w.__userflowStatePatched = false
+
+// patch the history API's
+// pushState method to emit userflow:pushstate and
+// replaceState method to emit userflow:replacestate
+const originalPushState = window.history.pushState
+const originalReplaceState = window.history.replaceState
```

Comment:
Use `w` here. It's for minification.

I think you could also assign `const history = w.history` at the top of this file (same place as `w = window`), and then refer to `history` instead of `w.history` everywhere else. We want the Userflow.js snippet to be as small as possible.